src/mi/Gaddress_validator.c

```diff
@@ -239,32 +236,37 @@ unw_address_is_valid(unw_word_t addr, size_t len)
   if (len == 0)
     return true;
 
-  if (unw_page_start (addr) == 0)
+  unw_word_t start_page_addr = unw_page_start (addr);
+  if (start_page_addr == 0)
     return false;
 
+  /*
+   * This should really bounds-check the end page address to avoid unsigned
+   * integer wraparound, but MAX_UNW_WORD_T is not a thing.
+   *
+   * if (addr < MAX_UNW_WORD_T - len - unw_page_size))
+   *   return false;
+   */
+  unw_word_t end_page_addr = unw_page_start (addr + (len - 1)) + unw_page_size;
+
   if (unlikely (!atomic_flag_test_and_set(&_unw_address_validator_initialized)))
     {
       _open_pipe ();
     }
 
-  unw_word_t lastbyte = addr + (len - 1); // highest addressed byte of data to access
-  while (1)
+  for (unw_word_t page_addr = start_page_addr;
+  	   page_addr < end_page_addr;
```

Comment:
if `end_page_addr` wrap around, then this will fail because we will not verify the page_addr. For example, page_addr = 2^64.

Reply:
@bregma unfortunately we are dealing with some weird bug with a JIT engine that output addresses like that, so running this uncover such edge cases.